### PR TITLE
Trend Micro Apex One Data Connector

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add Trend Micro Apex One Data Connector

### CODE REVIEW

**Comment by preetikr** [comment]
@YaronFruchtmann / @oshezaf - can you please review the parser to ensure aligns with the new normalization + KQL guidelines? Thanks.

**Review by v-rucdu — APPROVED** [review]

**Comment by v-rucdu** [comment]
@socprime : Please fix the failing validation checks

**Review by oshezaf — DISMISSED** [review]
@socprime: 
- Remove the all the column_ifexists, which are not necessary. Use project-rename when no transformation is needed.
- If we decide to move to ARM based deployment, this should be broken into multiple parsers per type. Each one is most commonly used alone and will fit differnt schemas as we move forward.
- If we don't, or until we do, use lookup/pack/back_unpack for the rename as it is much more performant than using union
- Try improving the normalized mapping based on the new guidelines. I am happy to assist.

File: Solutions/Trend Micro Apex One/Data Connectors/TrendMicro_ApexOne.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                    },
+                    "type": "CopyableLabel"
+                }
+            ]
+        },
+        {
+            "title": "4. Secure your machine ",
+            "description": "Make sure to configure the machine's security according to your organization's security policy\n\n\n[Learn more >](https://aka.ms/SecureCEF)"
+        }
+    ]
```
Comment (v-maudan) [inline Solutions/Trend Micro Apex One/Data Connectors/TrendMicro_ApexOne.json:125]: Please add a Metadata object within this file. See [https://github.com/Azure/Azure-Sentinel/blob/29c401e232cc7ce1ee91e7583133d3aa53e7e4af/DataConnectors/Templates/Connector_CEF_template.json](https://github.com/Azure/Azure-Sentinel/blob/29c401e232cc7ce1ee91e7583133d3aa53e7e4af/DataConnectors/Templates/Connector_CEF_template.json#L114) for further detail.

File: Solutions/Trend Micro Apex One/Data Connectors/TrendMicro_ApexOne.json [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    "connectivityCriterias": [
+        {
+            "type": "IsConnectedQuery",
+            "value": [
+                "TMApexOneEvent\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
+            ]
+        }
+    ],
+    "availability": {
+        "status": 1
```
Comment (v-jayakal) [inline Solutions/Trend Micro Apex One/Data Connectors/TrendMicro_ApexOne.json:35]: Can you please add "isPreview": true here?

**Review by v-jayakal — CHANGES_REQUESTED** [review]
@vu-socprime , Can you add metadata and update parser as per the comment provided by @oshezaf above please? Please let us know. IsPreview tag is the reason for validation failure.

**Comment by v-rucdu** [comment]
@socprime : Can you please resolve the review comments and Fix the Validation failures

Thanks!!!

File: Solutions/Trend Micro Apex One/Data Connectors/TrendMicro_ApexOne.json (hunk 2 above)
Comment (vu-socprime) [reply Solutions/Trend Micro Apex One/Data Connectors/TrendMicro_ApexOne.json]: updated

**Comment by v-rucdu** [comment]
@vu-socprime : Thanks for fixing the validation failures. 

Just wanted to know if you would like to go for the suggestions provided by Ofer as per new guidelines or you want us to merge these changes?

Thanks!!!

**Review by v-rucdu — APPROVED** [review]

**Comment by vu-socprime** [comment]
> @vu-socprime : Thanks for fixing the validation failures.
> 
> Just wanted to know if you would like to go for the suggestions provided by Ofer as per new guidelines or you want us to merge these changes?
> 
> Thanks!!!

@v-rucdu Parser updated

**Review by v-jayakal — APPROVED** [review]

**Review by v-rucdu — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**